File: src/main/java/ro/nicolaemariusghergu/queryit/model/Manufacturer.java
package ro.nicolaemariusghergu.queryit.model;

import lombok.*;
import org.hibernate.annotations.NotFound;
import org.hibernate.annotations.NotFoundAction;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;

@Getter
@Setter
@Entity
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "manufacturers")
public class Manufacturer implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NotFound(action = NotFoundAction.IGNORE)
    @Column(name = "manufacturer_id", nullable = false)
    private Long id;

    @Column(name = "name")
    private String name;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Manufacturer that = (Manufacturer) o;
        return id.equals(that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: src/main/kotlin/ro/nicolaemariusghergu/queryit/model/Manufacturer.kt
package ro.nicolaemariusghergu.queryit.model

import org.hibernate.annotations.NotFound
import org.hibernate.annotations.NotFoundAction
import javax.persistence.*

@Entity
@Table(name = "manufacturers")
class Manufacturer(
        @ManyToOne
        @JoinColumn(name = "deposit_id")
        @NotFound(action = NotFoundAction.IGNORE)
        val deposit: Deposit? = null,

        @ManyToOne
        @JoinColumn(name = "truck_id")
        @NotFound(action = NotFoundAction.IGNORE)
        val truck: Truck ? = null,

        @Column(name = "name")
        val name: String,

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @NotFound(action = NotFoundAction.IGNORE)
        @Column(name = "manufacturer_id", nullable = false)
        val id: Long ? = null
)